samples/brokers/onvif-video-broker/CredentialStore.cs

```diff
@@ -0,0 +1,198 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Json;
+
+namespace Akri 
+{
+	class Credential {
+		public string Username { get; set; }
+		public string Password { get; set; }
+	}
+
+	class CredentialData {
+    	public string Username { get; set; }
+    	public string Password { get; set; }
+    	public bool Base64encoded { get; set; }
+	}
+	
+	class CredentialRefData {
+    	public string Username_ref { get; set; }
+	    public string Password_ref { get; set; }
+	}
+
+	class CredentialStore {
+		private readonly string secretDirectory;
+		private readonly string configMapDirectory;
+
+		public CredentialStore(string secretDirectory, string configMapDirectory) {
+			this.secretDirectory = secretDirectory;
+			this.configMapDirectory = configMapDirectory;
+		}
+
+		public Credential Get(string id) {
+			if (String.IsNullOrEmpty(this.secretDirectory)) {
+				return null;
+			}
+
+			Credential defaultCredential = null;
+			// get credential from secrets
+			var (credential, isDefault) = GetCredentialFromSecret(id);
+			if ((credential != null) && !isDefault) {
+				return credential;
+			}
+			defaultCredential ??= credential;
+
+			// get credential from credential ref list
+			(credential, isDefault) = GetCredentialFromCredentialRefList(id);
+			if ((credential != null) && !isDefault) {
+				return credential;
+			}
+			defaultCredential ??= credential;
+
+			// get credential from credential list
+			(credential, isDefault) = GetCredentialFromCredentialList(id);
+			if ((credential != null) && !isDefault) {
+				return credential;
+			}
+			defaultCredential ??= credential;
+
+			return defaultCredential;
+		}
+
+		private (Credential, bool) GetCredentialFromCredentialRefList(string id) {
+			var credentialRefList = GetListContent<List<string>>("device_credential_ref_list");
+			if (credentialRefList == null) {
+				return (null, false);
+			}
+
+			var allRefDictionary = new Dictionary<string, CredentialRefData>();
+			foreach (var refList in credentialRefList) {
+				var refEntries = GetListContent<Dictionary<string, CredentialRefData>>(refList);
+				if (refEntries != null) {
+					refEntries.ToList().ForEach(x => allRefDictionary[x.Key] = x.Value);
+				}
+			}
+
+            var isDefault = false;
```

Comment:
nit: inconsistent indentation (maybe tabs instead of spaces?)

Reply:
use tabs